KimMiSeo/BOJ/BOJ17484.java

```diff
@@ -0,0 +1,117 @@
+
+
+import java.util.*;
+import java.io.*;
+
+public class BOJ17484 {
+	static int n,m;
+	static int graph[][];
+	static int min = 100*n*m;
+	
+	static int[] dy = {-1, 0, 1};
+	static int[] visited;
+//	static int result = 100*n*m;
+	
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		String[] temp = br.readLine().split(" ");
+		n = Integer.parseInt(temp[0]);
+		m = Integer.parseInt(temp[1]);
+		graph = new int[n][m];
+		
+		for (int i=0; i<n; i++) {
+			temp = br.readLine().split(" ");
+			for (int j=0; j<m; j++) {
+				graph[i][j] = Integer.parseInt(temp[j]);
+			}
+		}
+		
+//		for (int i=0; i<m; i++) {
+//			visited = new int[n];
+//			visited[0] = i;
+//			dfs(1,i,-1);
+//		}
+//		
+//		System.out.println(min);
+		
+		//--------- dp ----------		
+		int result = solve(graph);
+		System.out.println(result);
+		
+	}
+	
+	private static int solve(int[][] graph) {
+		int[][][] dp = new int[n][m][3];
+		
+		// 첫 줄 초기화
+		for (int i=0; i<m; i++) {
+			dp[0][i][0] = graph[0][i];
+			dp[0][i][1] = graph[0][i];
+			dp[0][i][2] = graph[0][i];
+			
+		}
+		
+		// dp 초기화
+		for (int i=1; i<n; i++) {
+			for (int j=0; j<m; j++) {
+				dp[i][j][0] = dp[i][j][1] = dp[i][j][2] = 100*n*m;
+			}
+		}
```

Comment:
위의 방식과 아래 방식 모두로 표기 가능할 때,
가독성이 어느 것이 좋다고 확답드리진 못하겠으나,
하나로 통일하는 것이 좋아 보입니다 !!


```suggestion
		for (int i=0; i<m; i++) {
			dp[0][i][0] = dp[0][i][1] = dp[0][i][2] = graph[0][i];
		}
		
		// dp 초기화
		for (int i=1; i<n; i++) {
			for (int j=0; j<m; j++) {
				dp[i][j][0] = dp[i][j][1] = dp[i][j][2] = 100*n*m;
			}
		}
```